### PR TITLE
Fix: Update TextInputTab to pass inputIndex to textChanged function, …

### PR DESCRIPTION
**Pull request by ShivanshShalabh**
…fixing URL overwirte with text inputs

### CODE REVIEW

File: src/components/Experiment/QuickInput/Tabs/TextInput/TextInputTab.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ export default function TextInputTab(props) {
             <textarea
                 value={text}
                 className={getElement("input")}
-                onChange={(e) => textChanged(e)}
+                onChange={(e) => textChanged(e, props?.inputIndex)}
```
Comment (walkingtowork) [inline src/components/Experiment/QuickInput/Tabs/TextInput/TextInputTab.js:18]: If this is a single input task, do we still have an `inputIndex`?

File: src/components/Experiment/QuickInput/Tabs/TextInput/useTextInputControl.js [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
             event.persist();
         }
 
         setText(event.target.value);
 
-        if (typeof(props.inputSelected) === "function") {
-            props.inputSelected(text, 0);
+        if (typeof (props.inputSelected) === "function") {
+            if (index === null) {
+                console.error("Index is required for multi-input tasks");
```
Comment (walkingtowork) [inline src/components/Experiment/QuickInput/Tabs/TextInput/useTextInputControl.js]: Asking the question above just to make sure that this error doesn't get thrown in single-input tasks.

Comment (ShivanshShalabh) [reply src/components/Experiment/QuickInput/Tabs/TextInput/useTextInputControl.js]: My bad, I thought ` if (typeof (props.inputSelected) === "function") ` is a check for multi-input. I have fixed that now.

File: src/components/Experiment/QuickInput/Tabs/TextInput/TextInputTab.js (hunk 1 above)
Comment (ShivanshShalabh) [reply src/components/Experiment/QuickInput/Tabs/TextInput/TextInputTab.js:18]: Yeah but it will be undefined so it will not matter. Should i change that?

Comment (walkingtowork) [reply src/components/Experiment/QuickInput/Tabs/TextInput/TextInputTab.js:18]: No, I guess it should be fine. I forgot that `undefined` and `null` aren't equivalent, so the code below should work for both single and multiinputs after all.

**Pull request merged**